aten/src/ATen/native/LossNLL.cpp

```diff
@@ -402,6 +417,70 @@ Tensor nll_loss(const Tensor & self, const Tensor & target, const Tensor & weigh
   return std::get<0>(at::nll_loss_forward(self, target, weight, reduction, ignore_index));
 }
 
+Tensor nll_loss_nd(
+    const Tensor& self,
+    const Tensor& target,
+    const Tensor& weight,
+    int64_t reduction,
+    int64_t ignore_index) {
+  if (self.dim() < 2) {
+    TORCH_CHECK(false, "Expected 2 or more dimensions (got ", self.dim(), ")");
+  }
+
+  if (self.sizes()[0] != target.sizes()[0]) {
+    TORCH_CHECK(
```

Comment:
You can use `TORCH_CHECK_VALUE` here to avoid having to change the tests

Reply:
That is super nice! I did not know about this. Will do!